src/exception.cr

```diff
@@ -129,6 +129,18 @@ class DivisionByZeroError < Exception
   end
 end
 
+# Raised when the result of an arithmetic operation is outside of the range
+# that can be represented within the given operands types.
+#
+# ```
+# Int32::Max + 1 # raises OverflowError (Overflow)
```

Comment:
What about adding an example about underflow?
```suggestion
# Int32::MAX + 1 # raises OverflowError (because of overflow)
# Int32::MIN - 0 # raises OverflowError (because of underflow)
```